wire-ios/Wire-iOS UnitTests/SystemSettingsTests.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+@testable import Wire
+
+final class SystemSettingsTests: XCTestCase {
+
+    func test_internalBuild() throws {
+        let url = try XCTUnwrap(Bundle.main.url(forResource: "Settings.bundle/Root", withExtension: "plist"))
+        let data = try Data(contentsOf: url)
+        let string = try XCTUnwrap(String(data: data, encoding: .utf8))
+
+        XCTAssertTrue(string.contains("DEVELOPER SETTINGS"))
+        XCTAssertTrue(string.contains("LICENSES"))
+    }
+
+    func test_productionBuild() {
+        // unit tests are always performed in internal builds,
+        // so we cannot assert production behavior.
+    }
```

Comment:
So there's no way to check that the licenses were added?

Reply:
Unfortunately not, at least I didn't find a way. @araneforseti told me that an integration test is not easy and we try to avoid it.
So either we have two options:
1. live with the behaviour as it is now - production builds will have no automated test.
2. perform the generation of licenses also on local machines and accept the additional time on the setup script. also after sdk, libs upgrade one needs to remember to re-run the setup script.

Please vote for 1 🎉 or 2 ❤️!
@caldrian @johnxnguyen @David-Henner @netbe @KaterinaWire @typfel @agisilaos

Reply:
For 2 you can add a danger rule to check the licenses where generated if deps changed

Reply:
Can a GitHub action check for uncommitted changes? Assuming that every PR runs the setup script, we could fail the build if we detect some uncommitted changes, such as those that come from generated license (i.e they weren't up to date). If that's the case, then option 2 would have a safe guard.

Reply:
To determine the right solution we need to know if the datadog license can be included in all cases or not.
If not, it would be dependent on environment vars, that has the consequence to generate on the CI every run.

will open a ticket and follow up PR for the datadog problem.